Sources/SwiftLexicalLookup/LookupConfig.swift

```diff
@@ -31,14 +33,17 @@
   /// If `finishInSequentialScope` would be set to `false`, the only name
   /// returned by lookup would be the `a` declaration from inside function body.
   @_spi(Experimental) public var finishInSequentialScope: Bool
+  @_spi(Experimental) public var buildConfiguration: BuildConfiguration?
 
   /// Creates a new lookup configuration.
   ///
   /// - `finishInSequentialScope` - specifies whether lookup should finish
   ///   in the closest sequential scope. `false` by default.
   @_spi(Experimental) public init(
-    finishInSequentialScope: Bool = false
+    finishInSequentialScope: Bool = false,
+    buildConfiguration: BuildConfiguration? = nil
```

Comment:
It's probably worth taking in a [`ConfiguredRegions`](https://swiftpackageindex.com/swiftlang/swift-syntax/main/documentation/swiftifconfig/configuredregions) here rather than a `BuildConfiguration`. `ConfiguredRegions` caches results so we aren't re-evaluating the same `#if`s repeatedly.